### PR TITLE
Feature/snippet routing

### PR DESCRIPTION
**Pull request by vrael560**

### CODE REVIEW

**Comment by vrael560** [comment]
@blimusiek @Vakrim wait for a while, I'm making huge refactor concerning new snippet dialog control...

File: client/js/stores/snippet-store.js [hunk 1]
```diff
@@ -5,7 +5,22 @@ import SnippetActions from '../actions/snippet-actions';
 class SnippetStore {
   constructor() {
     this.bindActions(SnippetActions);
-    this.snippets = [];
```
Comment (blimusiek) [inline client/js/stores/snippet-store.js]: @vrael560 you are removing here a this.snippet array. What is ok but ...

File: client/js/stores/snippet-store.js [hunk 2]
```diff
@@ -5,7 +5,22 @@ import SnippetActions from '../actions/snippet-actions';
...
+      this.setState({
+        snippets: snippets.concat(data.snippets)
+      });
+    } else {
+      // TODO react to errors
+      // console.log(data.error.message)
+    }
   }
 
   create(data) {
```
Comment (blimusiek) [inline client/js/stores/snippet-store.js]: @vrael560 in create ```this.snippet``` is still used. Please change it to use ```this.state``` instead

File: client/test/pages/snipetts-index.test.js [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+
+describe('SnippetsIndex', () => {
+  describe('add snippet', () => {
+    it('should open new snippet dialog on click', () => {
+      let snippetsIndex = TestUtils.renderIntoDocument(
+        <SnippetsIndex/>
+      );
+      expect(snippetsIndex.snippetDialog.dialog.isOpen()).to.be.false;
+      snippetsIndex.setState({dialogOpen: true});
+      expect(snippetsIndex.snippetDialog.dialog.isOpen()).to.be.true;
```
Comment (blimusiek) [inline client/test/pages/snipetts-index.test.js]: I think it doesn't check anything. You shoul use ```let b = /*find button node*/; TestUtils.Simulate.click(b);```

File: client/js/actions/snippet-actions.js [hunk 4]
```diff
@@ -8,10 +8,17 @@ class SnippetActions {
...
-  create(snippet) {
-    axios.post('/snippets', snippet)
-      .then(res => this.dispatch({ok: true, snippet: res.data}))
+  getAll () {
+    axios.get('/snippets')
+      .then(res => this.dispatch({ok: true, snippets: res.data}))
       .catch(err => this.dispatch({ok: false, error: err}));
   }
+
+  create (snippet) {
```
Comment (blimusiek) [inline client/js/actions/snippet-actions.js]: @vrael560 Why you comment here that ajax query?

File: client/test/pages/snipetts-index.test.js (hunk 3 above)
Comment (vrael560) [inline client/test/pages/snipetts-index.test.js]: It check state propagation to children, I think dialog has animation on open which delays change of state so assertion just after click fails

Comment (blimusiek) [inline client/test/pages/snipetts-index.test.js]: I will check it

**Comment by blimusiek** [comment]
It's ok for me

**Pull request merged**